src/phoebus_driver.cpp

```diff
@@ -1014,11 +1015,43 @@ TaskListStatus PhoebusDriver::MonteCarloStep() {
 
 /**
  * Gets called before output.
- * Currently: Fills tracers.
+ * Currently:
+ *  Fills Tracers
+ *  Computes entropy for output
  **/
 void UserWorkBeforeOutput(MeshBlock *pmb, ParameterInput *pin) {
   auto tracer_pkg = pmb->packages.Get("tracers");
   bool do_tracers = tracer_pkg->Param<bool>("active");
+
+  namespace p = fluid_prim;
+  auto rc = pmb->meshblock_data.Get().get();
+  PackIndexMap imap;
+  auto v = rc->PackVariables(
```

Comment:
Since we're writing new code here. Maybe use a sparse pack as we did in other places? Can you port this over to the newer design? Should work for both blocks and meshdata either way.

Reply:
Started with that and did not work. Everywhere else, I have MeshData rc, here the function does not get argument rc, I am accessing rc from pmb... maybe that's why, the error was it could not see p::variable.

Reply:
I think sparse packs should support `MeshBlockData` and `MeshData`...

Reply:
/home/mari/Work/Phoebus/phoebus/src/phoebus_driver.cpp: In lambda function:
/home/mari/Work/Phoebus/phoebus/src/phoebus_driver.cpp:1050:24: error: no match for call to ‘(const parthenon::SparsePack<fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye>) (fluid_prim::ye, const int&, const int&, const int&)’
 1050 |           lambda[0] = v(p::ye(), k, j, i);
      |                       ~^~~~~~~~~~~~~~~~~~
In file included from /home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/make_pack_descriptor.hpp:32,
                 from /home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/update.hpp:27,
                 from /home/mari/Work/Phoebus/phoebus/external/parthenon/src/parthenon/driver.hpp:23,
                 from /home/mari/Work/Phoebus/phoebus/src/phoebus_driver.cpp:22:
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:350:32: note: candidate: ‘template<class TIn, typename std::enable_if<IncludesType<TIn, fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye>::value, int>::type <anonymous> > auto& parthenon::SparsePack<Ts>::operator()(int, TE, const TIn&) const [with typename std::enable_if<IncludesType<TIn, Ts ...>::value, int>::type <anonymous> = TIn; Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}]’
  350 |   KOKKOS_INLINE_FUNCTION auto &operator()(const int b, const TE el, const TIn &t) const {
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:350:32: note:   template argument deduction/substitution failed:
/home/mari/Work/Phoebus/phoebus/src/phoebus_driver.cpp:1050:24: note:   candidate expects 3 arguments, 4 provided
 1050 |           lambda[0] = v(p::ye(), k, j, i);
      |                       ~^~~~~~~~~~~~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:356:32: note: candidate: ‘template<class TIn, typename std::enable_if<IncludesType<TIn, fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye>::value, int>::type <anonymous> > auto& parthenon::SparsePack<Ts>::operator()(int, const TIn&) const [with typename std::enable_if<IncludesType<TIn, Ts ...>::value, int>::type <anonymous> = TIn; Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}]’
  356 |   KOKKOS_INLINE_FUNCTION auto &operator()(const int b, const TIn &t) const {
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:356:32: note:   template argument deduction/substitution failed:
/home/mari/Work/Phoebus/phoebus/src/phoebus_driver.cpp:1050:24: note:   candidate expects 2 arguments, 4 provided
 1050 |           lambda[0] = v(p::ye(), k, j, i);
      |                       ~^~~~~~~~~~~~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:395:32: note: candidate: ‘template<class TIn, typename std::enable_if<IncludesType<TIn, fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye>::value, int>::type <anonymous> > parthenon::Real& parthenon::SparsePack<Ts>::operator()(int, const TIn&, int, int, int) const [with typename std::enable_if<IncludesType<TIn, Ts ...>::value, int>::type <anonymous> = TIn; Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}]’
  395 |   KOKKOS_INLINE_FUNCTION Real &operator()(const int b, const TIn &t, const int k,
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:395:32: note:   template argument deduction/substitution failed:
/home/mari/Work/Phoebus/phoebus/src/phoebus_driver.cpp:1050:24: note:   candidate expects 5 arguments, 4 provided
 1050 |           lambda[0] = v(p::ye(), k, j, i);
      |                       ~^~~~~~~~~~~~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:403:32: note: candidate: ‘template<class TIn, typename std::enable_if<IncludesType<TIn, fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye>::value, int>::type <anonymous> > parthenon::Real& parthenon::SparsePack<Ts>::operator()(int, TE, const TIn&, int, int, int) const [with typename std::enable_if<IncludesType<TIn, Ts ...>::value, int>::type <anonymous> = TIn; Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}]’
  403 |   KOKKOS_INLINE_FUNCTION Real &operator()(const int b, const TE el, const TIn &t,
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:403:32: note:   template argument deduction/substitution failed:
/home/mari/Work/Phoebus/phoebus/src/phoebus_driver.cpp:1050:24: note:   candidate expects 6 arguments, 4 provided
 1050 |           lambda[0] = v(p::ye(), k, j, i);
      |                       ~^~~~~~~~~~~~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:329:32: note: candidate: ‘auto& parthenon::SparsePack<Ts>::operator()(int, TE, int) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}; TE = parthenon::TopologicalElement]’
  329 |   KOKKOS_INLINE_FUNCTION auto &operator()(const int b, const TE el, const int idx) const {
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:329:32: note:   candidate expects 3 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:332:32: note: candidate: ‘auto& parthenon::SparsePack<Ts>::operator()(int, int) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}]’
  332 |   KOKKOS_INLINE_FUNCTION auto &operator()(const int b, const int idx) const {
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:332:32: note:   candidate expects 2 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:339:32: note: candidate: ‘auto& parthenon::SparsePack<Ts>::operator()(int, TE, parthenon::PackIdx) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}; TE = parthenon::TopologicalElement]’
  339 |   KOKKOS_INLINE_FUNCTION auto &operator()(const int b, const TE el, PackIdx idx) const {
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:339:32: note:   candidate expects 3 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:345:32: note: candidate: ‘auto& parthenon::SparsePack<Ts>::operator()(int, parthenon::PackIdx) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}]’
  345 |   KOKKOS_INLINE_FUNCTION auto &operator()(const int b, PackIdx idx) const {
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:345:32: note:   candidate expects 2 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:363:9: note: candidate: ‘parthenon::Real& parthenon::SparsePack<Ts>::operator()(int, int, int, int, int) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}; parthenon::Real = double]’
  363 |   Real &operator()(const int b, const int idx, const int k, const int j,
      |         ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:363:9: note:   candidate expects 5 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:369:9: note: candidate: ‘parthenon::Real& parthenon::SparsePack<Ts>::operator()(int, int, int, int) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}; parthenon::Real = double]’
  369 |   Real &operator()(int idx, const int k, const int j, const int i) const {
      |         ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:369:24: note:   no known conversion for argument 1 from ‘fluid_prim::ye’ to ‘int’
  369 |   Real &operator()(int idx, const int k, const int j, const int i) const {
      |                    ~~~~^~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:374:32: note: candidate: ‘parthenon::Real& parthenon::SparsePack<Ts>::operator()(int, TE, int, int, int, int) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}; parthenon::Real = double; TE = parthenon::TopologicalElement]’
  374 |   KOKKOS_INLINE_FUNCTION Real &operator()(const int b, const TE el, const int idx,
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:374:32: note:   candidate expects 6 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:379:32: note: candidate: ‘parthenon::Real& parthenon::SparsePack<Ts>::operator()(int, parthenon::PackIdx, int, int, int) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}; parthenon::Real = double]’
  379 |   KOKKOS_INLINE_FUNCTION Real &operator()(const int b, PackIdx idx, const int k,
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:379:32: note:   candidate expects 5 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:387:32: note: candidate: ‘parthenon::Real& parthenon::SparsePack<Ts>::operator()(int, TE, parthenon::PackIdx, int, int, int) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}; parthenon::Real = double; TE = parthenon::TopologicalElement]’
  387 |   KOKKOS_INLINE_FUNCTION Real &operator()(const int b, const TE el, PackIdx idx,
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:387:32: note:   candidate expects 6 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/src/phoebus_driver.cpp:1051:64: error: no match for call to ‘(const parthenon::SparsePack<fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye>) (fluid_prim::density, const int&, const int&, const int&)’
 1051 |           const Real s = eos_sc.EntropyFromDensityTemperature(v(p::density(), k, j, i), v(p::temperature(), k, j, i), lambda);
      |                                                               ~^~~~~~~~~~~~~~~~~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:350:32: note: candidate: ‘template<class TIn, typename std::enable_if<IncludesType<TIn, fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye>::value, int>::type <anonymous> > auto& parthenon::SparsePack<Ts>::operator()(int, TE, const TIn&) const [with typename std::enable_if<IncludesType<TIn, Ts ...>::value, int>::type <anonymous> = TIn; Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}]’
  350 |   KOKKOS_INLINE_FUNCTION auto &operator()(const int b, const TE el, const TIn &t) const {
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:350:32: note:   template argument deduction/substitution failed:
/home/mari/Work/Phoebus/phoebus/src/phoebus_driver.cpp:1051:64: note:   candidate expects 3 arguments, 4 provided
 1051 |           const Real s = eos_sc.EntropyFromDensityTemperature(v(p::density(), k, j, i), v(p::temperature(), k, j, i), lambda);
      |                                                               ~^~~~~~~~~~~~~~~~~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:356:32: note: candidate: ‘template<class TIn, typename std::enable_if<IncludesType<TIn, fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye>::value, int>::type <anonymous> > auto& parthenon::SparsePack<Ts>::operator()(int, const TIn&) const [with typename std::enable_if<IncludesType<TIn, Ts ...>::value, int>::type <anonymous> = TIn; Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}]’
  356 |   KOKKOS_INLINE_FUNCTION auto &operator()(const int b, const TIn &t) const {
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:356:32: note:   template argument deduction/substitution failed:
/home/mari/Work/Phoebus/phoebus/src/phoebus_driver.cpp:1051:64: note:   candidate expects 2 arguments, 4 provided
 1051 |           const Real s = eos_sc.EntropyFromDensityTemperature(v(p::density(), k, j, i), v(p::temperature(), k, j, i), lambda);
      |                                                               ~^~~~~~~~~~~~~~~~~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:395:32: note: candidate: ‘template<class TIn, typename std::enable_if<IncludesType<TIn, fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye>::value, int>::type <anonymous> > parthenon::Real& parthenon::SparsePack<Ts>::operator()(int, const TIn&, int, int, int) const [with typename std::enable_if<IncludesType<TIn, Ts ...>::value, int>::type <anonymous> = TIn; Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}]’
  395 |   KOKKOS_INLINE_FUNCTION Real &operator()(const int b, const TIn &t, const int k,
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:395:32: note:   template argument deduction/substitution failed:
/home/mari/Work/Phoebus/phoebus/src/phoebus_driver.cpp:1051:64: note:   candidate expects 5 arguments, 4 provided
 1051 |           const Real s = eos_sc.EntropyFromDensityTemperature(v(p::density(), k, j, i), v(p::temperature(), k, j, i), lambda);
      |                                                               ~^~~~~~~~~~~~~~~~~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:403:32: note: candidate: ‘template<class TIn, typename std::enable_if<IncludesType<TIn, fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye>::value, int>::type <anonymous> > parthenon::Real& parthenon::SparsePack<Ts>::operator()(int, TE, const TIn&, int, int, int) const [with typename std::enable_if<IncludesType<TIn, Ts ...>::value, int>::type <anonymous> = TIn; Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}]’
  403 |   KOKKOS_INLINE_FUNCTION Real &operator()(const int b, const TE el, const TIn &t,
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:403:32: note:   template argument deduction/substitution failed:
/home/mari/Work/Phoebus/phoebus/src/phoebus_driver.cpp:1051:64: note:   candidate expects 6 arguments, 4 provided
 1051 |           const Real s = eos_sc.EntropyFromDensityTemperature(v(p::density(), k, j, i), v(p::temperature(), k, j, i), lambda);
      |                                                               ~^~~~~~~~~~~~~~~~~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:329:32: note: candidate: ‘auto& parthenon::SparsePack<Ts>::operator()(int, TE, int) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}; TE = parthenon::TopologicalElement]’
  329 |   KOKKOS_INLINE_FUNCTION auto &operator()(const int b, const TE el, const int idx) const {
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:329:32: note:   candidate expects 3 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:332:32: note: candidate: ‘auto& parthenon::SparsePack<Ts>::operator()(int, int) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}]’
  332 |   KOKKOS_INLINE_FUNCTION auto &operator()(const int b, const int idx) const {
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:332:32: note:   candidate expects 2 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:339:32: note: candidate: ‘auto& parthenon::SparsePack<Ts>::operator()(int, TE, parthenon::PackIdx) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}; TE = parthenon::TopologicalElement]’
  339 |   KOKKOS_INLINE_FUNCTION auto &operator()(const int b, const TE el, PackIdx idx) const {
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:339:32: note:   candidate expects 3 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:345:32: note: candidate: ‘auto& parthenon::SparsePack<Ts>::operator()(int, parthenon::PackIdx) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}]’
  345 |   KOKKOS_INLINE_FUNCTION auto &operator()(const int b, PackIdx idx) const {
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:345:32: note:   candidate expects 2 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:363:9: note: candidate: ‘parthenon::Real& parthenon::SparsePack<Ts>::operator()(int, int, int, int, int) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}; parthenon::Real = double]’
  363 |   Real &operator()(const int b, const int idx, const int k, const int j,
      |         ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:363:9: note:   candidate expects 5 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:369:9: note: candidate: ‘parthenon::Real& parthenon::SparsePack<Ts>::operator()(int, int, int, int) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}; parthenon::Real = double]’
  369 |   Real &operator()(int idx, const int k, const int j, const int i) const {
      |         ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:369:24: note:   no known conversion for argument 1 from ‘fluid_prim::density’ to ‘int’
  369 |   Real &operator()(int idx, const int k, const int j, const int i) const {
      |                    ~~~~^~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:374:32: note: candidate: ‘parthenon::Real& parthenon::SparsePack<Ts>::operator()(int, TE, int, int, int, int) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}; parthenon::Real = double; TE = parthenon::TopologicalElement]’
  374 |   KOKKOS_INLINE_FUNCTION Real &operator()(const int b, const TE el, const int idx,
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:374:32: note:   candidate expects 6 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:379:32: note: candidate: ‘parthenon::Real& parthenon::SparsePack<Ts>::operator()(int, parthenon::PackIdx, int, int, int) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}; parthenon::Real = double]’
  379 |   KOKKOS_INLINE_FUNCTION Real &operator()(const int b, PackIdx idx, const int k,
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:379:32: note:   candidate expects 5 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:387:32: note: candidate: ‘parthenon::Real& parthenon::SparsePack<Ts>::operator()(int, TE, parthenon::PackIdx, int, int, int) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}; parthenon::Real = double; TE = parthenon::TopologicalElement]’
  387 |   KOKKOS_INLINE_FUNCTION Real &operator()(const int b, const TE el, PackIdx idx,
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:387:32: note:   candidate expects 6 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/src/phoebus_driver.cpp:1051:90: error: no match for call to ‘(const parthenon::SparsePack<fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye>) (fluid_prim::temperature, const int&, const int&, const int&)’
 1051 |           const Real s = eos_sc.EntropyFromDensityTemperature(v(p::density(), k, j, i), v(p::temperature(), k, j, i, lambda);
      |                                                                                         ~^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:350:32: note: candidate: ‘template<class TIn, typename std::enable_if<IncludesType<TIn, fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye>::value, int>::type <anonymous> > auto& parthenon::SparsePack<Ts>::operator()(int, TE, const TIn&) const [with typename std::enable_if<IncludesType<TIn, Ts ...>::value, int>::type <anonymous> = TIn; Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}]’
  350 |   KOKKOS_INLINE_FUNCTION auto &operator()(const int b, const TE el, const TIn &t) const {
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:350:32: note:   template argument deduction/substitution failed:
/home/mari/Work/Phoebus/phoebus/src/phoebus_driver.cpp:1051:90: note:   candidate expects 3 arguments, 4 provided
 1051 |           const Real s = eos_sc.EntropyFromDensityTemperature(v(p::density(), k, j, i), v(p::temperature(), k, j, i, lambda);
      |                                                                                         ~^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:356:32: note: candidate: ‘template<class TIn, typename std::enable_if<IncludesType<TIn, fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye>::value, int>::type <anonymous> > auto& parthenon::SparsePack<Ts>::operator()(int, const TIn&) const [with typename std::enable_if<IncludesType<TIn, Ts ...>::value, int>::type <anonymous> = TIn; Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}]’
  356 |   KOKKOS_INLINE_FUNCTION auto &operator()(const int b, const TIn &t) const {
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:356:32: note:   template argument deduction/substitution failed:
/home/mari/Work/Phoebus/phoebus/src/phoebus_driver.cpp:1051:90: note:   candidate expects 2 arguments, 4 provided
 1051 |           const Real s = eos_sc.EntropyFromDensityTemperature(v(p::density(), k, j, i), v(p::temperature(), k, j, i, lambda);
      |                                                                                         ~^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:395:32: note: candidate: ‘template<class TIn, typename std::enable_if<IncludesType<TIn, fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye>::value, int>::type <anonymous> > parthenon::Real& parthenon::SparsePack<Ts>::operator()(int, const TIn&, int, int, int) const [with typename std::enable_if<IncludesType<TIn, Ts ...>::value, int>::type <anonymous> = TIn; Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}]’
  395 |   KOKKOS_INLINE_FUNCTION Real &operator()(const int b, const TIn &t, const int k,
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:395:32: note:   template argument deduction/substitution failed:
/home/mari/Work/Phoebus/phoebus/src/phoebus_driver.cpp:1051:90: note:   candidate expects 5 arguments, 4 provided
 1051 |           const Real s = eos_sc.EntropyFromDensityTemperature(v(p::density(), k, j, i), v(p::temperature(), k, j, i, lambda);
      |                                                                                         ~^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:403:32: note: candidate: ‘template<class TIn, typename std::enable_if<IncludesType<TIn, fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye>::value, int>::type <anonymous> > parthenon::Real& parthenon::SparsePack<Ts>::operator()(int, TE, const TIn&, int, int, int) const [with typename std::enable_if<IncludesType<TIn, Ts ...>::value, int>::type <anonymous> = TIn; Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}]’
  403 |   KOKKOS_INLINE_FUNCTION Real &operator()(const int b, const TE el, const TIn &t,
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:403:32: note:   template argument deduction/substitution failed:
/home/mari/Work/Phoebus/phoebus/src/phoebus_driver.cpp:1051:90: note:   candidate expects 6 arguments, 4 provided
 1051 |           const Real s = eos_sc.EntropyFromDensityTemperature(v(p::density(), k, j, i), v(p::temperature(), k, j, i, lambda);
      |                                                                                         ~^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:329:32: note: candidate: ‘auto& parthenon::SparsePack<Ts>::operator()(int, TE, int) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}; TE = parthenon::TopologicalElement]’
  329 |   KOKKOS_INLINE_FUNCTION auto &operator()(const int b, const TE el, const int idx) const {
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:329:32: note:   candidate expects 3 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:332:32: note: candidate: ‘auto& parthenon::SparsePack<Ts>::operator()(int, int) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}]’
  332 |   KOKKOS_INLINE_FUNCTION auto &operator()(const int b, const int idx) const {
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:332:32: note:   candidate expects 2 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:339:32: note: candidate: ‘auto& parthenon::SparsePack<Ts>::operator()(int, TE, parthenon::PackIdx) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}; TE = parthenon::TopologicalElement]’
  339 |   KOKKOS_INLINE_FUNCTION auto &operator()(const int b, const TE el, PackIdx idx) const {
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:339:32: note:   candidate expects 3 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:345:32: note: candidate: ‘auto& parthenon::SparsePack<Ts>::operator()(int, parthenon::PackIdx) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}]’
  345 |   KOKKOS_INLINE_FUNCTION auto &operator()(const int b, PackIdx idx) const {
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:345:32: note:   candidate expects 2 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:363:9: note: candidate: ‘parthenon::Real& parthenon::SparsePack<Ts>::operator()(int, int, int, int, int) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}; parthenon::Real = double]’
  363 |   Real &operator()(const int b, const int idx, const int k, const int j,
      |         ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:363:9: note:   candidate expects 5 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:369:9: note: candidate: ‘parthenon::Real& parthenon::SparsePack<Ts>::operator()(int, int, int, int) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}; parthenon::Real = double]’
  369 |   Real &operator()(int idx, const int k, const int j, const int i) const {
      |         ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:369:24: note:   no known conversion for argument 1 from ‘fluid_prim::temperature’ to ‘int’
  369 |   Real &operator()(int idx, const int k, const int j, const int i) const {
      |                    ~~~~^~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:374:32: note: candidate: ‘parthenon::Real& parthenon::SparsePack<Ts>::operator()(int, TE, int, int, int, int) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}; parthenon::Real = double; TE = parthenon::TopologicalElement]’
  374 |   KOKKOS_INLINE_FUNCTION Real &operator()(const int b, const TE el, const int idx,
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:374:32: note:   candidate expects 6 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:379:32: note: candidate: ‘parthenon::Real& parthenon::SparsePack<Ts>::operator()(int, parthenon::PackIdx, int, int, int) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}; parthenon::Real = double]’
  379 |   KOKKOS_INLINE_FUNCTION Real &operator()(const int b, PackIdx idx, const int k,
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:379:32: note:   candidate expects 5 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:387:32: note: candidate: ‘parthenon::Real& parthenon::SparsePack<Ts>::operator()(int, TE, parthenon::PackIdx, int, int, int) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}; parthenon::Real = double; TE = parthenon::TopologicalElement]’
  387 |   KOKKOS_INLINE_FUNCTION Real &operator()(const int b, const TE el, PackIdx idx,
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:387:32: note:   candidate expects 6 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/src/phoebus_driver.cpp:1052:12: error: no match for call to ‘(const parthenon::SparsePack<fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye>) (fluid_prim::entropy, const int&, const int&, const int&)’
 1052 |           v(p::entropy(), k, j, i) = s;
      |           ~^~~~~~~~~~~~~~~~~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:350:32: note: candidate: ‘template<class TIn, typename std::enable_if<IncludesType<TIn, fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye>::value, int>::type <anonymous> > auto& parthenon::SparsePack<Ts>::operator()(int, TE, const TIn&) const [with typename std::enable_if<IncludesType<TIn, Ts ...>::value, int>::type <anonymous> = TIn; Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}]’
  350 |   KOKKOS_INLINE_FUNCTION auto &operator()(const int b, const TE el, const TIn &t) const {
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:350:32: note:   template argument deduction/substitution failed:
/home/mari/Work/Phoebus/phoebus/src/phoebus_driver.cpp:1052:12: note:   candidate expects 3 arguments, 4 provided
 1052 |           v(p::entropy(), k, j, i) = s;
      |           ~^~~~~~~~~~~~~~~~~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:356:32: note: candidate: ‘template<class TIn, typename std::enable_if<IncludesType<TIn, fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye>::value, int>::type <anonymous> > auto& parthenon::SparsePack<Ts>::operator()(int, const TIn&) const [with typename std::enable_if<IncludesType<TIn, Ts ...>::value, int>::type <anonymous> = TIn; Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}]’
  356 |   KOKKOS_INLINE_FUNCTION auto &operator()(const int b, const TIn &t) const {
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:356:32: note:   template argument deduction/substitution failed:
/home/mari/Work/Phoebus/phoebus/src/phoebus_driver.cpp:1052:12: note:   candidate expects 2 arguments, 4 provided
 1052 |           v(p::entropy(), k, j, i) = s;
      |           ~^~~~~~~~~~~~~~~~~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:395:32: note: candidate: ‘template<class TIn, typename std::enable_if<IncludesType<TIn, fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye>::value, int>::type <anonymous> > parthenon::Real& parthenon::SparsePack<Ts>::operator()(int, const TIn&, int, int, int) const [with typename std::enable_if<IncludesType<TIn, Ts ...>::value, int>::type <anonymous> = TIn; Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}]’
  395 |   KOKKOS_INLINE_FUNCTION Real &operator()(const int b, const TIn &t, const int k,
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:395:32: note:   template argument deduction/substitution failed:
/home/mari/Work/Phoebus/phoebus/src/phoebus_driver.cpp:1052:12: note:   candidate expects 5 arguments, 4 provided
 1052 |           v(p::entropy(), k, j, i) = s;
      |           ~^~~~~~~~~~~~~~~~~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:403:32: note: candidate: ‘template<class TIn, typename std::enable_if<IncludesType<TIn, fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye>::value, int>::type <anonymous> > parthenon::Real& parthenon::SparsePack<Ts>::operator()(int, TE, const TIn&, int, int, int) const [with typename std::enable_if<IncludesType<TIn, Ts ...>::value, int>::type <anonymous> = TIn; Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}]’
  403 |   KOKKOS_INLINE_FUNCTION Real &operator()(const int b, const TE el, const TIn &t,
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:403:32: note:   template argument deduction/substitution failed:
/home/mari/Work/Phoebus/phoebus/src/phoebus_driver.cpp:1052:12: note:   candidate expects 6 arguments, 4 provided
 1052 |           v(p::entropy(), k, j, i) = s;
      |           ~^~~~~~~~~~~~~~~~~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:329:32: note: candidate: ‘auto& parthenon::SparsePack<Ts>::operator()(int, TE, int) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}; TE = parthenon::TopologicalElement]’
  329 |   KOKKOS_INLINE_FUNCTION auto &operator()(const int b, const TE el, const int idx) const {
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:329:32: note:   candidate expects 3 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:332:32: note: candidate: ‘auto& parthenon::SparsePack<Ts>::operator()(int, int) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}]’
  332 |   KOKKOS_INLINE_FUNCTION auto &operator()(const int b, const int idx) const {
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:332:32: note:   candidate expects 2 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:339:32: note: candidate: ‘auto& parthenon::SparsePack<Ts>::operator()(int, TE, parthenon::PackIdx) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}; TE = parthenon::TopologicalElement]’
  339 |   KOKKOS_INLINE_FUNCTION auto &operator()(const int b, const TE el, PackIdx idx) const {
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:339:32: note:   candidate expects 3 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:345:32: note: candidate: ‘auto& parthenon::SparsePack<Ts>::operator()(int, parthenon::PackIdx) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}]’
  345 |   KOKKOS_INLINE_FUNCTION auto &operator()(const int b, PackIdx idx) const {
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:345:32: note:   candidate expects 2 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:363:9: note: candidate: ‘parthenon::Real& parthenon::SparsePack<Ts>::operator()(int, int, int, int, int) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}; parthenon::Real = double]’
  363 |   Real &operator()(const int b, const int idx, const int k, const int j,
      |         ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:363:9: note:   candidate expects 5 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:369:9: note: candidate: ‘parthenon::Real& parthenon::SparsePack<Ts>::operator()(int, int, int, int) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}; parthenon::Real = double]’
  369 |   Real &operator()(int idx, const int k, const int j, const int i) const {
      |         ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:369:24: note:   no known conversion for argument 1 from ‘fluid_prim::entropy’ to ‘int’
  369 |   Real &operator()(int idx, const int k, const int j, const int i) const {
      |                    ~~~~^~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:374:32: note: candidate: ‘parthenon::Real& parthenon::SparsePack<Ts>::operator()(int, TE, int, int, int, int) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}; parthenon::Real = double; TE = parthenon::TopologicalElement]’
  374 |   KOKKOS_INLINE_FUNCTION Real &operator()(const int b, const TE el, const int idx,
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:374:32: note:   candidate expects 6 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:379:32: note: candidate: ‘parthenon::Real& parthenon::SparsePack<Ts>::operator()(int, parthenon::PackIdx, int, int, int) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}; parthenon::Real = double]’
  379 |   KOKKOS_INLINE_FUNCTION Real &operator()(const int b, PackIdx idx, const int k,
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:379:32: note:   candidate expects 5 arguments, 4 provided
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:387:32: note: candidate: ‘parthenon::Real& parthenon::SparsePack<Ts>::operator()(int, TE, parthenon::PackIdx, int, int, int) const [with Ts = {fluid_prim::density, fluid_prim::entropy, fluid_prim::temperature, fluid_prim::ye}; parthenon::Real = double; TE = parthenon::TopologicalElement]’
  387 |   KOKKOS_INLINE_FUNCTION Real &operator()(const int b, const TE el, PackIdx idx,
      |                                ^~~~~~~~
/home/mari/Work/Phoebus/phoebus/external/parthenon/src/interface/sparse_pack.hpp:387:32: note:   candidate expects 6 arguments, 4 provided
make[2]: *** [src/CMakeFiles/phoebus.dir/build.make:748: src/CMakeFiles/phoebus.dir/phoebus_driver.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1804: src/CMakeFiles/phoebus.dir/all] Error 2
make: *** [Makefile:146: all] Error 2

Reply:
Looks like the reason for the crash was not having a block index as the first arg to the pack? e.g., doing `v(0, p::density, k, j, i)` gets rid of the crash. Do the sparse packs not support not passing that arg? Or is it set up wrong here? (I hid your comment @mari2895 because it was really long, sorry)

Reply:
OK let me get all fixed and commit, meanwhile I almost added cs and divv/cs so it will be in this PR as well.